partiql-parser/src/main/kotlin/org/partiql/parser/internal/V1PartiQLParserDefault.kt

```diff
@@ -0,0 +1,2195 @@
+import org.partiql.parser.internal.antlr.PartiQLParser
+import org.partiql.parser.internal.antlr.PartiQLParserBaseVisitor
+import org.partiql.parser.internal.util.DateTimeUtils
+import org.partiql.spi.Context
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.PErrorKind
+import org.partiql.spi.errors.PErrorListener
+import org.partiql.spi.errors.PErrorListenerException
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.boolValue
+import org.partiql.value.dateValue
+import org.partiql.value.datetime.DateTimeException
+import org.partiql.value.datetime.DateTimeValue
+import org.partiql.value.decimalValue
+import org.partiql.value.int32Value
+import org.partiql.value.int64Value
+import org.partiql.value.intValue
+import org.partiql.value.missingValue
+import org.partiql.value.nullValue
+import org.partiql.value.stringValue
+import org.partiql.value.timeValue
+import org.partiql.value.timestampValue
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.math.MathContext
+import java.math.RoundingMode
+import java.nio.channels.ClosedByInterruptException
+import java.nio.charset.StandardCharsets
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+import org.partiql.parser.internal.antlr.PartiQLParser as GeneratedParser
+import org.partiql.parser.internal.antlr.PartiQLTokens as GeneratedLexer
+
+/**
+ * ANTLR Based Implementation of a PartiQLParser
+ *
+ * SLL Prediction Mode
+ * -------------------
+ * The [PredictionMode.SLL] mode uses the [BailErrorStrategy]. The [GeneratedParser], upon seeing a syntax error,
+ * will throw a [ParseCancellationException] due to the [GeneratedParser.getErrorHandler]
+ * being a [BailErrorStrategy]. The purpose of this is to throw syntax errors as quickly as possible once encountered.
+ * As noted by the [PredictionMode.SLL] documentation, to guarantee results, it is useful to follow up a failed parse
+ * by parsing with [PredictionMode.LL]. See the JavaDocs for [PredictionMode.SLL] and [BailErrorStrategy] for more.
+ *
+ * LL Prediction Mode
+ * ------------------
+ * The [PredictionMode.LL] mode is capable of parsing all valid inputs for a grammar,
+ * but is slower than [PredictionMode.SLL]. Upon seeing a syntax error, this parser throws a [PartiQLParserException].
+ */
+internal class V1PartiQLParserDefault : V1PartiQLParser {
+
+    @Throws(PErrorListenerException::class)
+    override fun parse(source: String, ctx: Context): V1PartiQLParser.Result {
+        try {
+            return parse(source, ctx.errorListener)
+        } catch (e: PErrorListenerException) {
+            throw e
+        } catch (throwable: Throwable) {
+            val error = PError.INTERNAL_ERROR(PErrorKind.SYNTAX(), null, throwable)
+            ctx.errorListener.report(error)
+            return V1PartiQLParser.Result.empty(source)
+        }
+    }
+
+    companion object {
```

Comment:
These were just copied from the existing `PartiQLParserDefault` for context. Since the static methods within the companion object are all private, I will just move them out of the companion object.